app/src/components/integrations/elements/pluginIcon/pluginIcon.jsx

```diff
@@ -24,39 +24,59 @@ import { globalIntegrationsSelector } from 'controllers/plugins/selectors';
 import { filterIntegrationsByName, isPluginSupportsCommonCommand } from 'controllers/plugins/utils';
 import { PLUGIN_DEFAULT_IMAGE, PLUGIN_IMAGES_MAP } from 'components/integrations/constants';
 import { Image } from 'components/main/image';
+import { PUBLIC_PLUGIN_ACCESS_TYPE } from 'controllers/plugins/constants';
 
 export const PluginIcon = ({ pluginData, className, ...rest }) => {
   const { details, name } = pluginData;
   const isDynamicIconAvailable = details && details.binaryData && details.binaryData.icon;
   const projectId = useSelector(activeProjectSelector);
   const globalIntegrations = useSelector(globalIntegrationsSelector);
 
-  const calculateIconSrc = () => {
+  const isPublic = details && details.accessType === PUBLIC_PLUGIN_ACCESS_TYPE;
+
+  const calculateIconParams = () => {
+    const putRequestParams = { method: 'PUT', data: { fileKey: 'icon' } };
```

Comment:
Please rename to smth like `commandParams` as it is a specific to commands API.
And move it inside of if block.